DBAgenda.php

```diff
@@ -22,6 +22,55 @@ public function __construct(string $CalDAV_url, string $CalDAV_username, string
     abstract public function createDB();
     abstract protected function openDB();
 
+    public function clean_orphan_categories($quiet = false) {
+        $sql = "FROM categories WHERE not exists (
+                select 1
+                from events_categories
+                where events_categories.category_id = categories.id
+        );";
+
+        if(!$quiet) {
+            $query = $this->pdo->prepare("SELECT * " . $sql);
+            $query->execute();
+            $results = $query->fetchAll(\PDO::FETCH_UNIQUE|\PDO::FETCH_ASSOC);
+            foreach($results as $id => $category) {
+                $this->log->info("Category $category[name] will be deleted.");
+            }
+        }
+        $query = $this->pdo->prepare("DELETE " . $sql);
+        $query->execute();
+        $this->log->info("Cleaning orphan categories - done.");
+    }
+    
+    public function clean_orphan_attendees($quiet = true) {
+        $sql = "FROM attendees WHERE not exists (
```

Comment:
for the same reason as `clean_orphan_categories` I think this should be in a transaction (to ensure we print exactly what we delete)